subprojects/core/src/main/java/org/gradle/internal/scripts/BuildScriptCompileUnitOfWork.java

```diff
@@ -76,7 +89,10 @@ public Identity identify(Map<String, ValueSnapshot> identityInputs, Map<String,
     @Override
     public WorkOutput execute(ExecutionRequest executionRequest) {
         File workspace = executionRequest.getWorkspace();
-        compileTo(classesDir(workspace));
+        File output = compileTo(originalDir(workspace));
+        // We should instrument output of classes directory to the classes directory instead of a jar,
+        // but currently instrumentation classloader doesn't support that yet.
+        instrument(output, transformedJar(workspace));
```

Comment:
:question: For me `transformed` was quite confusing in this context. Shall we call it `instrumented` instead?

Reply:
Changed and we now use `instrumented`